contributing/docs-style.md

```diff
@@ -0,0 +1,379 @@
+    <td>### Heading</td>
+    <td><center>Level-3</center></td>
+    <td rowspan="2" style="vertical-align : middle;">Sub-sections. Used in each Section to separate supporting information or tasks.</td>
+  </tr>
+  <tr>
+    <td>#### Heading</td>
+    <td><center>Level-4</center></td>
+  </tr>
+</thead>
+</table>
+</font>
+
+<div class="alert alert-warning">
+  <h3>Best Practice</h3>
+  Use descriptive language in the phrasing of headings. For example, for a documentation page on dynamic tables, "Dynamic Tables and Continuous Queries" is more descriptive than "Background" or "Technical Information".
+</div>
+
+### Table of Contents
+
+In the documentation build, the **Table Of Contents** (TOC) is automatically generated from the headings of the page using the following line of markup:
+
+```liquid
+{{ "{% toc " }}%}
+```
+
+All headings up to **Level-3** are considered. To exclude a particular heading from the TOC:
+
+```liquid
+{{ "# Excluded Heading
+    {:.no_toc" }}}
+```
+
+<div class="alert alert-warning">
+  <h3>Best Practice</h3>
+  Write a short and concise introduction to the topic that is being covered and place it before the TOC. A little context, such an outline of key messages, goes a long way in ensuring that the documentation is consistent and accessible to all levels of knowledge.
+</div>
+
+### Navigation
+
+In the documentation build, navigation is defined using properties configured in the front-matter variables of each page.
+
+<font size="3">
+<table width="100%" class="table table-bordered">
+  <thead>
+  <tr>
+    <th></th>
+    <th style="vertical-align : middle;"><center><b>Variable</b></center></th>
+    <th style="vertical-align : middle;"><center><b>Possible Values</b></center></th>
+    <th style="vertical-align : middle;"><center><b>Description</b></center></th>
+  </tr>
+  <tr>
+    <td rowspan="4" style="vertical-align : middle;"><b>Navigation</b></td>
+    <td>nav-id</td>
+    <td>%s</td>
+    <td>The ID of the page. Other pages can use this ID as their nav-parent_id.</td>
+  </tr>
+  <tr>
+    <td>nav-parent_id</td>
+    <td>{root,%s}</td>
+    <td>The ID of the parent page. The lowest navigation level is root.</td>
+  </tr>
+  <tr>
+    <td>nav-pos</td>
+    <td>%d</td>
+    <td>The relative position of pages per navigation level.</td>
+  </tr>
+  <tr>
+    <td>nav-title</td>
+    <td>%s</td>
+    <td>The title to use as an override of the default link text (title).</td>
+  </tr>
+ </thead>
+</table>
+</font>
+
+It's possible to use _Back to Top_ links in extensive documentation pages, so that users can navigate to the top of the page without having to scroll back up manually. In markup, this is implemented as a placeholder that is replaced with a default link when generating the documentation:
+
+```liquid
+{{ "{% top " }}%}
+```
+
+<div class="alert alert-warning">
+  <h3>Best Practice</h3>
+  It's recommended to use Back to Top links at least at the end of each Level-2 section.
+</div>
+
+### Annotations
```

Comment:
This means we want to move away completely from the block-style highlight blocks that are used, for example, in https://ci.apache.org/projects/flink/flink-docs-master/dev/stream/state/broadcast_state.html?